From d6d622cc287acb6cda523232c9dcae7d099ba10a Mon Sep 17 00:00:00 2001
From: Abdur Rehman <abdur_rehman@mentor.com>
Date: Wed, 24 Aug 2016 20:12:13 +0500
Subject: [PATCH] rpmsg_imx6q_driver: store master cpu id

Store cpu id that is required by some MEMF tracepoints.

Signed-off-by: Yasir-Khan <yasir_khan@mentor.com>
Signed-off-by: Abdur Rehman <abdur_rehman@mentor.com>
---
 drivers/rpmsg/rpmsg_imx6q_driver.c    | 2 ++
 drivers/rpmsg/rpmsg_imx6q_internals.h | 1 +
 2 files changed, 3 insertions(+)

diff --git a/drivers/rpmsg/rpmsg_imx6q_driver.c b/drivers/rpmsg/rpmsg_imx6q_driver.c
index 3024d04..1019ed7 100644
--- a/drivers/rpmsg/rpmsg_imx6q_driver.c
+++ b/drivers/rpmsg/rpmsg_imx6q_driver.c
@@ -355,6 +355,8 @@ static int imx6q_rpmsg_initialize(struct platform_device *pdev)
 	virtio_dev->config	  = &imx6q_rpmsg_virtio_config_ops;
 	virtio_dev->dev.parent  = &(imx6q_rpmsg_p->mid_dev);
 	virtio_dev->dev.release = imx6q_rpmsg_vdev_release;
+	imx6q_rpmsg_p->master_cpu_id = MASTER_CPU_ID;
+	virtio_dev->priv	= &(imx6q_rpmsg_p->master_cpu_id);
 
 	/* Register the virtio device. */
 	ret = register_virtio_device(virtio_dev);
diff --git a/drivers/rpmsg/rpmsg_imx6q_internals.h b/drivers/rpmsg/rpmsg_imx6q_internals.h
index c7671fc..145d4b7 100644
--- a/drivers/rpmsg/rpmsg_imx6q_internals.h
+++ b/drivers/rpmsg/rpmsg_imx6q_internals.h
@@ -54,6 +54,7 @@ struct imx6q_rpmsg_instance {
 	u32	 virtioid;
 	u32	 ringtx;
 	u32	 ringrx;
+	u32	 master_cpu_id;
 
 	struct virtio_device virtio_dev;
 
-- 
2.8.1

